src/ui/drawing/ISurface.hh

```diff
@@ -76,7 +76,7 @@ public:
     /// <param name="nWidth">The width to draw the image.</param>
     /// <param name="nHeight">The height to draw the image.</param>
     /// <param name="pImage">The image to draw.</param>
-    virtual void DrawImage(int nX, int nY, int nWidth, int nHeight, const ImageReference& pImage) = 0;
+    virtual void DrawImage(int nX, int nY, int nWidth, int nHeight, _In_ ImageReference& pImage) = 0;
```

Comment:
I'm not totally sure what you want me to do with this. Making it const will require `const_cast` which usually leads to undefined behavior.

Reply:
I want `ImageReference` to remain const. The `ImageReference` itself is immutable, but `ImageRepository::GetHBitmap` stores the `HBITMAP` in `ImageReference.m_nData` to avoid doing future lookups. `GetData`/`SetData` are only used by `GetHBitmap` and should probably be hidden from the public interface. In C#, I'd use an explicit interface implementation to get to the data member, but I don't think C++ has the equivalent functionality. Maybe they could be made private and exposed via `friend` like this:
```
private:
    unsigned long GetData() const noexcept { return m_nData; }
    void SetData(unsigned long nValue) const noexcept { m_nData = nValue; }
    mutable unsigned long m_nData{};
    friend class ra::ui::drawing::gdi::ImageRepository;
```
Unless you have a better idea?

Reply:
Your idea sounds good to me. I did have an idea a long time ago to explicitly implement some of the WIC interfaces but that's too Windows Centric.

Reply:
I'll just look at this on the weekend or something.

Reply:
done, had to make GDISurface a friend as well. Also needed to overload GetData.